M6-Backend/MockFacility.test.js

```diff
@@ -0,0 +1,210 @@
+//Test Set 1 --> Testing the interface specific for module facility
+const interfacespecific = require('./interfacespecific');
+
+test("interfacespecific invalid input", ()=>{
+  expect(interfacespecific("ll@@@@}}")).not.toEqual('posts');
+})
+
+test("interfacespecific number out of bound", ()=>{
+  expect(interfacespecific(100000)).not.toEqual('studys');
+})
+
+test("interfacespecific inbound wrong type", ()=>{
+  expect(interfacespecific(1)).not.toEqual('posts');
+})
+
+test("interfacespecific posts success", ()=>{
+  expect(interfacespecific(0)).toEqual('posts');
+})
+
+test("interfacespecific studys success", ()=>{
+  expect(interfacespecific(1)).toEqual('studys');
+})
+
+test("interfacespecific entertainments success", ()=>{
+  expect(interfacespecific(2)).toEqual('entertainments');
+})
+
+test("interfacespecific restaurants success", ()=>{
+  expect(interfacespecific(3)).toEqual('restaurants');
+})
+
+
+
+//Test 2--> Test set for rate interface
+const interfaceRate = require('./interfaceRate');
+
+test("interfacesRate invalid input", ()=>{
+  expect(interfaceRate("ll@@@@}}")).not.toEqual(0);
+})
+
+test("empty rate", ()=>{
+  expect(interfaceRate([])).toEqual(0);
+})
+
+test("interfacesRate valid array invalid result", ()=>{
+  expect(interfaceRate([1,2,3])).not.toEqual(3);
+})
+
+test("interfacesRate valid array and valid result", ()=>{
+  expect(interfaceRate([1,2,3,4,5,6,7,8])).toEqual(4.5);
+})
+
+
+//Test set 3 -->   Test set for newest
+const interfaceNewest = require('./interfaceNewestFacility');
+
+test("interfaceNewest invalid input", ()=>{
+  expect(interfaceNewest("ll@@@@}}")).not.toEqual({result:"",length:0});
+})
+
+test("interfaceNewest empty input", ()=>{
+  expect(interfaceNewest([])).toEqual({result:"",length:0});
+})
+
+test("interfaceNewest valid input invalid result", ()=>{
+  expect(interfaceNewest(["3","3","3"])).not.toEqual({result:"",length:5});
+})
+
+test("interfaceNewest valid input and valid result", ()=>{
```

Comment:
Codacy has a fix for the issue: [Replace `=>` with `·=>·`](https://app.codacy.com/gh/HizanLi/CPEN321/pullRequest?prid=9816566)
```suggestion
test("interfaceNewest valid input and valid result", () => {
```